ecs-cli/modules/cli/servicediscovery/servicediscovery_helper.go

```diff
@@ -0,0 +1,290 @@
+		return fmt.Errorf("container_port is a required field when using SRV DNS records")
+	}
+
+	hasPublic := hasNamespace(input.PublicDNSNamespace.Namespace)
+	hasPrivate := hasNamespace(input.PrivateDNSNamespace.Namespace)
+
+	if hasPublic && hasPrivate {
+		return fmt.Errorf("both a public and private namespace can not be used with Service Discovery; please use only 1 namespace")
+	}
+
+	if !hasPublic && !hasPrivate {
+		return fmt.Errorf("to use Service Discovery, please specify a dns namespace")
+	}
+
+	if input.PrivateDNSNamespace.Name != "" && input.PrivateDNSNamespace.ID == "" {
+		if input.PrivateDNSNamespace.VPC == "" {
+			return fmt.Errorf("vpc is required when specifying private namespace by name")
+		}
+	}
+
+	return nil
+}
+
+func resolveNamespaceOverride(namespaceNameFromFlag, namespaceIDFromFlag, namespaceType string, ecsParamsNamespace utils.Namespace) utils.Namespace {
+	flagNamespace := utils.Namespace{
+		ID:   namespaceIDFromFlag,
+		Name: namespaceNameFromFlag,
+	}
+	if hasNamespace(flagNamespace) && hasNamespace(ecsParamsNamespace) {
+		showFieldOverrideMsg(getNamespace(ecsParamsNamespace), getNamespace(ecsParamsNamespace), fmt.Sprintf("dns %s namespace", namespaceType))
+	}
+	if hasNamespace(flagNamespace) {
+		return flagNamespace
+	}
+	return ecsParamsNamespace
+}
+
+// Validates that namespace name and ID fields were not both specified
+// This function runs before flags and ECS Params are merged
+func validateNotNameAndID(c *cli.Context, ecsParamsSD *utils.ServiceDiscovery) error {
+	if c.String(flags.PrivateDNSNamespaceIDFlag) != "" && c.String(flags.PrivateDNSNamespaceNameFlag) != "" {
+		return fmt.Errorf("Validation Error: %s and %s both specified", flags.PrivateDNSNamespaceIDFlag, flags.PrivateDNSNamespaceNameFlag)
+	}
+
+	if c.String(flags.PublicDNSNamespaceIDFlag) != "" && c.String(flags.PublicDNSNamespaceNameFlag) != "" {
+		return fmt.Errorf("Validation Error: %s and %s both specified", flags.PublicDNSNamespaceIDFlag, flags.PublicDNSNamespaceNameFlag)
+	}
+
+	if ecsParamsSD.PrivateDNSNamespace.Name != "" && ecsParamsSD.PrivateDNSNamespace.ID != "" {
+		return fmt.Errorf("Validation Error: private_dns_namespace.name and private_dns_namespace.id both specified")
+	}
+
+	if ecsParamsSD.PublicDNSNamespace.Name != "" && ecsParamsSD.PublicDNSNamespace.ID != "" {
+		return fmt.Errorf("Validation Error: public_dns_namespace.name and public_dns_namespace.id both specified")
+	}
+
+	return nil
+}
+
+// Logs warnings for fields which are ignored when a namespace ID is specified
+func namespaceWarningsWhenIDSpecified(input *utils.ServiceDiscovery) {
+	msg := "Ignoring %s because the ID of an existing private namespace was specified"
+
+	privNamespace := input.PrivateDNSNamespace
+	if privNamespace.ID != "" {
+		if privNamespace.Description != "" {
+			logrus.Warnf(msg, "description")
+		}
+		if privNamespace.VPC != "" {
+			logrus.Warnf(msg, "vpc")
+		}
+	}
+}
+
+func hasNamespace(n utils.Namespace) bool {
```

Comment:
Could this be simplifed to `return  n.ID != "" || n.Name != ""`?

Reply:
Good point